pkgs/tools/security/cryptomator/default.nix

```diff
@@ -0,0 +1,113 @@
+    owner = "cryptomator";
+    repo = "cryptomator";
+    rev = "${version}";
+    sha256 = "1v7s3y0dh00vbi4zdlg3msv3034sa5qzx5fq5xwi76vs2r37nk2j";
+  };
+
+  desktopItem = makeDesktopItem {
+    name = "cryptomator";
+    comment = "Cloud Storage Encryption Utility";
+    desktopName = "Cryptomator";
+    type = "Application";
+    exec = "cryptomator";
+    icon = "org.cryptomator.Cryptomator";
+    mimeType="application/x-vnd.cryptomator-vault-metadata";
+    categories = "Utility;Security;FileTools;";
+    terminal = "false";
+    extraEntries = ''
+      StartupWMClass=org-cryptomator-launcher-Cryptomator
+    '';
+  };
+
+  # perform fake build to make a fixed-output derivation out of the files downloaded from maven central (120MB)
+  deps = stdenv.mkDerivation {
+    name = "cryptomator-${version}-deps";
+    inherit src;
+
+    buildInputs = [ openjdk maven ];
+
+    buildPhase = ''
+      cd main
+      while mvn -Prelease package -Dmaven.repo.local=$out/.m2 -Dmaven.wagon.rto=5000; [ $? = 1 ]; do
+        echo "timeout, restart maven to continue downloading"
+      done
+    '';
+
+    # keep only *.{pom,jar,sha1,nbm} and delete all ephemeral files with lastModified timestamps inside
+    installPhase = ''
+      find $out/.m2 -type f -regex '.+\(\.lastUpdated\|resolver-status\.properties\|_remote\.repositories\)' -delete
+      find $out/.m2 -type f -iname '*.pom' -exec sed -i -e 's/\r\+$//' {} \;
+    '';
+
+    outputHashAlgo = "sha256";
+    outputHashMode = "recursive";
+    outputHash = "15gwldn9dc4gxi8ja4lay5qdadvhxry17ggj3cqamk8g38rh97v0";
+  };
+
+in stdenv.mkDerivation rec {
+  inherit pname;
+  inherit version;
+  inherit src;
+
+  buildPhase = ''
+    cd main
+    mvn -Prelease package --offline -Dmaven.repo.local=$(cp -dpR ${deps}/.m2 ./ && chmod +w -R .m2 && pwd)/.m2
+  '';
+
+  installPhase = ''
+    mkdir -p $out/usr/share/cryptomator/libs/
+    mkdir -p $out/bin/
+
+    cp buildkit/target/libs/* buildkit/target/linux-libs/* $out/usr/share/cryptomator/libs/
+
+    cat > $out/usr/share/cryptomator/cryptomator <<EOF
+    #!/usr/bin/env sh
+    exec ${openjdk}/bin/java -verbose \
+        -cp "$out/usr/share/cryptomator/libs/*" \
+        -Dcryptomator.settingsPath="~/.config/Cryptomator/settings.json" \
+        -Dcryptomator.ipcPortPath="~/.config/Cryptomator/ipcPort.bin" \
+        -Dcryptomator.logDir="~/.local/share/Cryptomator/logs" \
+        -Dcryptomator.mountPointsDir="~/.local/share/Cryptomator/mnt" \
+        -Djdk.gtk.version=2 \
+        -Xss20m \
+        -Xmx512m \
+        org.cryptomator.launcher.Cryptomator
+EOF
+
+    chmod +x $out/usr/share/cryptomator/cryptomator
+    ln -s $out/usr/share/cryptomator/cryptomator $out/bin/cryptomator
+
+    wrapProgram "$out/usr/share/cryptomator/cryptomator" \
+     --prefix PATH : "$out/usr/share/cryptomator/libs/:${lib.makeBinPath [ openjdk glib ]}" \
+     --prefix LD_LIBRARY_PATH : "${lib.makeLibraryPath [ fuse ]}" \
+     --set JAVA_HOME "$openjdk"
```

Comment:
Does `"$openjdk"` even work here (quoted and `$openjdk` normally shouldn't be set though I'm not familiar with our Java abstractions). Maybe this was supposed to be `${openjdk}`.

Also: I'd expect that OpenJDK is only a build-time dependency and the JRE is enough during run-time but in this case I wouldn't object to using OpenJDK, at least for now (seems like packaging this was already painful enough).

Reply:
I also think that JRE should be enough. Do you know how to get it? I didn't see derivation for just jre.

Reply:
I had a look at `pkgs/top-level/all-packages.nix` and it seems like things have changed (my bad :o):
```
  /* default JDK */

  jdk = jdk15;

  # Since the introduction of the Java Platform Module System in Java 9, Java
  # no longer ships a separate JRE package.
  #
  # If you are building a 'minimal' system/image, you are encouraged to use
  # 'jre_minimal' to build a bespoke JRE containing only the modules you need.
  #
  # For a general-purpose system, 'jre' defaults to the full JDK:
  jre = jdk15;
  jre_headless = jdk15_headless;

  jre_minimal = callPackage ../development/compilers/openjdk/jre.nix { };

  openjdk = openjdk15;
  openjdk_headless = openjdk15_headless;
```

Not sure if using `jre` (e.g. to make it explicit in the code) or `openjdk` is preferred but it doesn't change the result so it should be fine either way.

The manual doesn't seem to cover such details: https://nixos.org/manual/nixpkgs/unstable/#sec-language-java
But `--set JAVA_HOME ${jdk.home}` seems useful (the `.home` part).

Reply:
Thanks, I will have a look. Conceptually I think we should use `jre` even if it is the same at the moment.

Reply:
Using `${jre.home}` now